theano/gof/vm.py

```diff
@@ -890,6 +894,94 @@ def make_all(self, profiler=None, input_storage=None,
             compute_map[k] = [k.owner is None]
 
         thunks = []
+
```

Comment:
To help make the code more redable and more maintanable, I think all this code block and the one starting at line 1004, could be moved into a function in this file. What do you think of this?

After that, I'll need to run the tests here before merging.